Files/Filesystem/WSLDistroManager.cs

```diff
@@ -1,140 +1,86 @@
 ﻿using Files.DataModels.NavigationControlItems;
-using Files.UserControls;
 using Microsoft.Toolkit.Mvvm.ComponentModel;
 using Microsoft.Toolkit.Uwp;
 using System;
 using System.Collections.ObjectModel;
 using System.Linq;
+using System.Collections.Generic;
 using System.Threading.Tasks;
-using Windows.ApplicationModel.Core;
 using Windows.Storage;
-using Windows.UI.Core;
 
 namespace Files.Filesystem
 {
     public class WSLDistroManager : ObservableObject
     {
+        public event EventHandler<List<INavigationControlItem>> RefreshCompleted;
+        public event EventHandler RemoveWslSidebarSection;
+
+        public LocationItem WslSection = null;
+
         public WSLDistroManager()
         {
         }
 
         public async Task EnumerateDrivesAsync()
         {
-            try
+            var distroFolder = await StorageFolder.GetFolderFromPathAsync(@"\\wsl$\");
+            if ((await distroFolder.GetFoldersAsync()).Count != 0)
             {
-                await SyncSideBarItemsUI();
-            }
-            catch (Exception) // UI Thread not ready yet, so we defer the pervious operation until it is.
-            {
-                System.Diagnostics.Debug.WriteLine($"RefreshUI Exception");
-                // Defer because UI-thread is not ready yet (and DriveItem requires it?)
-                CoreApplication.MainView.Activated += EnumerateDrivesAsync;
-            }
-        }
-
-        private async void EnumerateDrivesAsync(CoreApplicationView sender, Windows.ApplicationModel.Activation.IActivatedEventArgs args)
-        {
-            await SyncSideBarItemsUI();
-            CoreApplication.MainView.Activated -= EnumerateDrivesAsync;
-        }
+                if (App.AppSettings.ShowWslSection && WslSection == null)
+                {
+                    WslSection = new LocationItem()
```

Comment:
Why is WSL manager different from the others? It seems to be the only onw to create it's own sidebar section

Reply:
@gave92 WslDistroManager used to be a lot worse before the work with this PR. Oh, and thanks for improving upon that :)

Reply:
Np, indeed there wasn't much left to do for WslDistroManager 👍